src/main/java/courseManager/InstanceList.java

```diff
@@ -0,0 +1,143 @@
+package courseManager;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class InstanceList {
+    private ArrayList<AssessmentInstance> listOfAssessmentInstances;
+
+    private String pluralTitle;
+    private String singularTitle;
+    private int totalNumberOfInstances;
+    private int numberOfCommittedInstances;
+    private int numberOfSubmittedInstances;
+
+    /**
+     * Create a new InstanceList object with a list of AssessmentInstances
+     * @param pluralTitle the plural title of the assessment
+     *              e.g. "Quizzes", "Midterm", "Homework Assignments"
+     * @param totalNumberOfInstances the total number of instances of the assessment
+     */
+
+    public InstanceList(String pluralTitle, int totalNumberOfInstances) {
+        this.listOfAssessmentInstances = new ArrayList<>();
+        this.singularTitle = toSingular(pluralTitle);
+        for (int i = 0, j = 1; i < totalNumberOfInstances; i++, j++){
+            this.addInstance(this.singularTitle + " #" + j);
+        }
+    }
+
+    public String toSingular(String pluralTitle){
+        if (pluralTitle == "Quizzes"){
+            return "Quiz";
+        }
+        if (pluralTitle.charAt(pluralTitle.length()-1) == 's'){
+            return singularTitle.substring(0, pluralTitle.length()-1);
+        }
+        return singularTitle;
+    }
+
+    public int getTotalNumberOfInstances() {
+        return totalNumberOfInstances;
+    }
+
+    public int getNumberOfCommittedInstances() {
+        int numberOfCommittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isCommitted()) {
+                numberOfCommittedInstances++;
+            }
+        }
+        return numberOfCommittedInstances;
+    }
+
+    public int getNumberOfSubmittedInstances() {
+        int numberOfSubmittedInstances = 0;
+        for (AssessmentInstance instance : listOfAssessmentInstances) {
+            if (instance.isSubmitted()) {
+                numberOfSubmittedInstances++;
+            }
+        }
+        return numberOfSubmittedInstances;
+    }
+
+    public double[] getAllMarks() {
+        double[] allMarks = new double[totalNumberOfInstances];
+        for (int i = 0; i < totalNumberOfInstances; i++) {
+            allMarks[i] = listOfAssessmentInstances.get(i).getMark();
+        }
```

Comment:
2 Issues with this. Firstly, as mentioned in my comment in AssessmentInstance, marks can be Null which could complicate thing for a function trying to work on them. Secondly, remember that the WeightScheme operates on marks that have been submitted "so far", which means that if a mark here is not given, we'd need to exclude it from the array. 

P.S. this function would need to be modified if you were to add the second constructor I discussed above.

Reply:
that's a good catch with the null again! i'll account for that
I wrote this function for computeHypothethicalGrade, because this version of computing the course's grade doesn't care whether a mark is submitted or committed, it takes all that were set to a double and that are not null. 
To account for that, I added a filter to the stream that checks for whether there is a mark (I transformed null into -1.0 so I didn't have to deal with weird autoboxing/unboxing). You will see in the pull request I will make soon and check it out!